rpc/codec/wire.pb/Makefile

```diff
@@ -15,14 +15,17 @@
 #
 # Author: Spencer Kimball (spencer.kimball@gmail.com)
 
-PROTOC      := ../../../../../../../bin/protoc
+# Ensure we only have one entry in GOPATH
+GOPATH := $(shell pwd)/../../../../../../..
+
+PROTOC      := $(GOPATH)/bin/protoc
 PROTOS      := $(wildcard *.proto)
 PROTO_GO    := $(PROTOS:.proto=.pb.go)
-GOGO_PROTOS := ../../../../github.com/gogo/protobuf/gogoproto/gogo.proto
+GOGO_PROTOS := $(GOPATH)/src/github.com/gogo/protobuf/gogoproto/gogo.proto
 
-PROTO_PATH := ../../../../../../:../../../../../../github.com/gogo/protobuf/protobuf:../../../../../../github.com/gogo/protobuf/gogoproto
+PROTO_PATH := $(GOPATH)/src/github.com/gogo/protobuf:$(GOPATH)/src/github.com/gogo/protobuf/protobuf
 
 all: $(PROTO_GO)
-
+.PHONY: $(PROTO_GO)
```

Comment:
This will ensure that the protos are regenerated on every call to make, even if the associated `.proto` files have not changed. Not sure if that is what Tamir intended or not. Seems reasonable, though, given that we invoke `make` and `go generate` rarely. I'd simplify this whole thing as:

```
.PHONY: all
all:
    $(PROTOC) ...
```